### DISCUSSION TITLE
[Feature] Store app-id + title and automatically set default-column-width to last used column width

### DISCUSSION THREAD

**Discussion by phuhl** [YaLTeR/niri#568] [Ideas]
Similar to how  MS Windows works (if I remember correctly).

- When resizing a window, niri would memorize this and, next time that I open said window, set the window to last used width (within bounds, if I accidentally resized to 0, I would not want the new window to be 0 px wide again)
  - i.e. automatic default-column-width definition upon resizing a window
- Open Question: How to identify windows?
  - By title? By app-id? By combo? Anything else?
  - I don't want Firefox to be resized based on the Firefox file picker width, etc. -> not only app-id

Benefits of this approach:
- No need to manually code window-rules for this
- "Automagically" supports a change in my workflow. When I decide one day, I want window X to be of size Y, niri will understand and give me window X in size Y automatically, next time

**Comment by YaLTeR** [comment]
Identifying windows is a problem, yeah. We could maybe do something like, restore column width window rule, where you match things explicitly, and that's what it saves/restores. You could then do a match-all with exclude rules.

I don't think it's possible to do this matching robustly automatically. I know some apps will save-restore window size on their own, but I'm not sure how well they work when niri requests just one dimension to be set (default-column-width 0).

**Reply by phuhl** [reply]
We could also communicate default width and height... 

Btw, is there a reason why we don't (yet) have a default-height feature? One can set min-/max-height but that also constraints resizing the window in that dimension.

**Reply by YaLTeR** [reply]
> Btw, is there a reason why we don't (yet) have a default-height feature?

I guess it's not really that useful in a tiling layout. With floating it'll be more useful.
